virtcontainers/acrn.go

```diff
@@ -0,0 +1,630 @@
+		return err
+	}
+
+	memory, err := a.memoryTopology()
+	if err != nil {
+		return err
+	}
+
+	kernelPath, err := a.config.KernelAssetPath()
+	if err != nil {
+		return err
+	}
+
+	initrdPath, err := a.config.InitrdAssetPath()
+	if err != nil {
+		return err
+	}
+
+	kernel := Kernel{
+		Path:       kernelPath,
+		InitrdPath: initrdPath,
+		Params:     a.kernelParameters(),
+	}
+
+	// Disabling UUID check until the below is fixed.
+	// https://github.com/kata-containers/runtime/issues/1785
+	/*
+		if a.state.UUID == "" {
+			return fmt.Errorf("UUID should not be empty")
+		}
+	*/
+
+	devices, err := a.buildDevices(initrdPath)
+	if err != nil {
+		return err
+	}
+
+	acrnPath, err := a.acrnPath()
+	if err != nil {
+		return err
+	}
+
+	acrnctlPath, err := a.acrnctlPath()
+	if err != nil {
+		return err
+	}
+
+	acrnConfig := Config{
+		ACPIVirt: true,
+		Path:     acrnPath,
+		CtlPath:  acrnctlPath,
+		Memory:   memory,
+		NumCPU:   a.config.NumVCPUs,
+		Devices:  devices,
+		Kernel:   kernel,
+		Name:     fmt.Sprintf("sandbox-%s", a.id),
+	}
+
+	a.acrnConfig = acrnConfig
+
+	return nil
+}
+
+// startSandbox will start the Sandbox's VM.
+func (a *acrn) startSandbox(timeout int) error {
+	span, _ := a.trace("startSandbox")
+	defer span.Finish()
+
+	if a.config.Debug {
+		params := a.arch.kernelParameters(a.config.Debug)
+		strParams := SerializeParams(params, "=")
+		formatted := strings.Join(strParams, " ")
+
+		// The name of this field matches a similar one generated by
+		// the runtime and allows users to identify which parameters
+		// are set here, which come from the runtime and which are set
+		// by the user.
+		a.Logger().WithField("default-kernel-parameters", formatted).Debug()
+	}
+
+	vmPath := filepath.Join(store.RunVMStoragePath, a.id)
+	err := os.MkdirAll(vmPath, store.DirMode)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			if err := os.RemoveAll(vmPath); err != nil {
+				a.Logger().WithError(err).Error("Fail to clean up vm directory")
+			}
+		}
+	}()
+
+	var strErr string
+	var PID int
+	PID, strErr, err = LaunchAcrn(a.acrnConfig, virtLog.WithField("subsystem", "acrn-dm"))
+	if err != nil {
+		return fmt.Errorf("%s", strErr)
+	}
+	a.info.PID = PID
+
+	if err = a.waitSandbox(timeout); err != nil {
+		a.Logger().WithField("Acrn wait failed:", err).Debug()
+		return err
+	}
+
+	//Store VMM information
+	return a.store.Store(store.Hypervisor, a.info)
+
+}
+
+// waitSandbox will wait for the Sandbox's VM to be up and running.
+func (a *acrn) waitSandbox(timeout int) error {
+	span, _ := a.trace("waitSandbox")
+	defer span.Finish()
+
+	if timeout < 0 {
+		return fmt.Errorf("Invalid timeout %ds", timeout)
+	}
+
+	time.Sleep(time.Duration(timeout) * time.Second)
+
+	return nil
+}
+
+// stopSandbox will stop the Sandbox's VM.
+func (a *acrn) stopSandbox() (err error) {
+
+	span, _ := a.trace("stopSandbox")
+	defer span.Finish()
+
+	a.Logger().Infof("Stopping ACRN VM")
```

Comment:
Nit: (Here and elsewhere) - You don't need to use the `f` version of this call as you are passing a string with no formats inside it:

```go
a.Logger().Info("Stopping ACRN VM")
```

Reply:
done.